src/context/GameContext.js

```diff
@@ -0,0 +1,168 @@
+const { createContext, useState, useContext, useEffect } = require('react');
+
+const GameContext = createContext();
+const GameProvider = ({ children }) => {
+  const startingBoard = [
+    { id: 0, content: '' },
+    { id: 1, content: '' },
+    { id: 2, content: '' },
+    { id: 3, content: '' },
+    { id: 4, content: '' },
+    { id: 5, content: '' },
+    { id: 6, content: '' },
+    { id: 7, content: '' },
+    { id: 8, content: '' },
+  ];
+  const [boardState, setBoardState] = useState(startingBoard);
+  const [winner, setWinner] = useState('');
+  const [draw, setDraw] = useState(false);
+  const [playerTurn, setPlayerTurn] = useState('X');
+  const [gameMessage, setGameMessage] = useState(`It's time for '${playerTurn}' to make a move`);
+  const [gameOver, setGameOver] = useState(false);
+
+  useEffect(() => {
+    const gameCheck = () => {
+      if (draw) {
+        setGameMessage("It's a draw! Better luck next time!");
+      } else {
+        winner && gameOver
+          ? setGameMessage(`'${winner}' wins the round!`)
+          : setGameMessage(`It's time for '${playerTurn}' to make a move`);
+      }
+    };
+    gameCheck();
+  }, [playerTurn, winner, gameOver, draw]);
+
+  function handleResetButton() {
+    setBoardState(startingBoard);
+    setWinner();
+    setDraw(false);
+    setPlayerTurn('X');
+    setGameMessage(`It's time for '${playerTurn}' to make a move`);
+    setGameOver(false);
+  }
+
+  function handleClick(content, id) {
+    if (content !== '') return;
+    if (gameOver === true) return;
+    // I can't think of a way to use setBoardState that would be dryer than this.
+    boardState[id].content = playerTurn;
```

Comment:
This is fine since but if you were using the board state in multiple places around the app you should be using the setState.  You can actually alter the board and then just setBoard to the current board as a way to circumnavigate this.